docs/how_to/compose_quantum_circuits.rst

```diff
@@ -0,0 +1,185 @@
+#####################
+Join quantum circuits
+#####################
+
+This guide shows how to combine different :class:`~.QuantumCircuit` objects.
+
+Build the circuits
+==================
+
+The first step is creating the circuits you want to combine.
+
+.. testcode::
+
+    from qiskit import QuantumCircuit
+
+    qc1 = QuantumCircuit(2,1)
+    qc1.h(0)
+    qc1.cx(0,1)
+    qc1.measure(1,0)
+
+    qc2 = QuantumCircuit(4,2)
+    qc2.y(0)
+    qc2.x(1)
+    qc2.cx(0,3)
+    qc2.measure(3,1)
+
+    print(qc1.draw()) 
+    print(qc2.draw())
+
+.. testoutput::
+    :options: +NORMALIZE_WHITESPACE
+
+         ┌───┐        
+    q_0: ┤ H ├──■─────
+         └───┘┌─┴─┐┌─┐
+    q_1: ─────┤ X ├┤M├
+              └───┘└╥┘
+    c: 1/═══════════╩═
+                    0 
+         ┌───┐        
+    q_0: ┤ Y ├──■─────
+         ├───┤  │     
+    q_1: ┤ X ├──┼─────
+         └───┘  │     
+    q_2: ───────┼─────
+              ┌─┴─┐┌─┐
+    q_3: ─────┤ X ├┤M├
+              └───┘└╥┘
+    c: 2/═══════════╩═
+                    1 
+
+Combine the circuits
+====================
+
+Now that you have built the circuits, they can be combined with two different methods:
+
+* :meth:`~.QuantumCircuit.compose`
+* :meth:`~.QuantumCircuit.append`
+
+.. warning::
+
+     The usage of the ``+`` operator to combine :class:`~.QuantumCircuit`\ s is no longer supported so if you try to define
+     a circuit as ``composed_qc = qc1 + qc2`` you will get an error.
+
+One detail these two methods have in common is that if the circuits have different sizes, they have to be applied to the one that has the most of both qubits and classical bits.
+
+:meth:`~.QuantumCircuit.compose`
+------------------------------------------------
+
+In order to combine two circuits with :meth:`~.QuantumCircuit.compose`, you only have to specify the circuit you want to insert. That way the qubits and bits of the smaller circuit will be included into the first qubits and bits of the bigger one in the original order they had. 
+
+By default, :meth:`~.QuantumCircuit.compose` does not change the circuit to which it is applied but returns the composed circuit. This can be changed by setting the ``inplace`` argument to ``True``.
+
+.. testcode::
+
+    qc3 = qc2.compose(qc1)
+    print(qc3.draw())
+
+.. testoutput::
+    :options: +NORMALIZE_WHITESPACE
+
+         ┌───┐     ┌───┐        
+    q_0: ┤ Y ├──■──┤ H ├──■─────
+         ├───┤  │  └───┘┌─┴─┐┌─┐
+    q_1: ┤ X ├──┼───────┤ X ├┤M├
+         └───┘  │       └───┘└╥┘
+    q_2: ───────┼─────────────╫─
+              ┌─┴─┐ ┌─┐       ║ 
+    q_3: ─────┤ X ├─┤M├───────╫─
+              └───┘ └╥┘       ║ 
+    c: 2/════════════╩════════╩═
+                     1        0 
+
+If you want to insert the qubits and bits into specific positions in the bigger circuit, you can use the ``qubits`` and ``bits`` arguments.
```

Comment:
Can you add a sentence to explain that the smaller circuit is now joint in the following way: q0 => q3 and q1=>q1 (smaller circuit index => bigger circuit index).

Reply:
Addressed in https://github.com/Qiskit/qiskit-terra/pull/8624/commits/802b53dbd8c534821d49284accd74403c8886dee